.circleci/config.yml

```diff
@@ -1,112 +1,48 @@
       - shellcheck/check:
           dir: ./src/scripts
           exclude: SC2148
-      # optional: Run BATS tests against your scripts
+          filters: *all-branches-and-tags
       - bats/run:
           path: ./src/tests
-      # Publish development version(s) of the orb.
-      - orb-tools/publish-dev:
+          filters: *all-branches-and-tags
+
+      # Publish a development version of the orb.
+      - orb-tools/publish:
           orb-name: datadog/synthetics-ci-orb
-          context: orb-publishing # A restricted context containing your private publishing credentials. Will only execute if approved by an authorized user.
+          vcs-type: << pipeline.project.type >>
+          context: orb-publishing # A restricted context containing private publishing credentials. Will only execute if approved by an authorized user.
           requires:
             - orb-tools/lint
             - orb-tools/pack
-            - bats/run
+            - orb-tools/review
             - shellcheck/check
-      # Trigger an integration workflow to test the
-      # dev:${CIRCLE_SHA1:0:7} version of your orb
-      - orb-tools/trigger-integration-tests-workflow:
-          name: trigger-integration-dev
-          context: orb-publishing
-          requires:
-            - orb-tools/publish-dev
+            - bats/run
+          filters: *all-branches-and-tags
 
-  # This `integration-test_deploy` workflow will only run
-  # when the run-integration-tests pipeline parameter is set to true.
-  # It is meant to be triggered by the "trigger-integration-tests-workflow"
-  # job, and run tests on <your orb>@dev:${CIRCLE_SHA1:0:7}.
-  integration-test_deploy:
-    when: << pipeline.parameters.run-integration-tests >>
-    jobs:
-      # Run any integration tests defined within the `jobs` key.
-      - integration-test-simple
-      - integration-test-with-spaces
-      - integration-test-with-multiple-public-ids
-      # Publish a semver version of the orb. relies on
-      # the commit subject containing the text "[semver:patch|minor|major|skip]"
-      # as that will determine whether a patch, minor or major
-      # version will be published or if publishing should
-      # be skipped.
-      # e.g. [semver:patch] will cause a patch version to be published.
-      - orb-tools/dev-promote-prod-from-commit-subject:
-          orb-name: datadog/synthetics-ci-orb
-          context: orb-publishing
-          add-pr-comment: false
-          fail-if-semver-not-indicated: true
-          publish-version-tag: false
+      # Triggers the next workflow in the Orb Development Kit.
+      - orb-tools/continue:
+          pipeline-number: << pipeline.number >>
+          vcs-type: << pipeline.project.type >>
           requires:
-            - integration-test-simple
-            - integration-test-with-spaces
-            - integration-test-with-multiple-public-ids
-          filters:
-            branches:
-              only:
-                - main
```

Comment:
This is replaced by a tag-based approach: we'll need to create a tag `vX.Y.Z` to publish the production orb.